test/websocket.js

```diff
@@ -52,3 +52,18 @@ test('basic websocket proxy', async (t) => {
     server.close()
   ])
 })
+
+test('captures errors on start', async (t) => {
+  const app = Fastify()
+  await app.listen(0)
+
+  const app2 = Fastify()
+  app2.register(proxy, { upstream: 'http://localhost', websocket: true })
+
+  const appPort = app.server.address().port
+
+  await t.rejects(app2.listen(appPort), /EADDRINUSE/)
```

Comment:
I am a little lost on this test. I don't think it's the websocket server that errors, but fastify itself. Can you check/confirm?

Reply:
You can run this test on `master`: it won't pass because Websocket server will cause `unhandledRejection` and crash test suite